src/main/java/edu/kis/vh/nursery/list/datastructures/IntArrayStack.java

```diff
@@ -0,0 +1,80 @@
+package edu.kis.vh.nursery.list.datastructures;
+
+import edu.kis.vh.nursery.Stack;
+
+public class IntArrayStack implements Stack {
+
+    private static final int MAX_SIZE = 12;
+
+    private final int[] numbers = new int[MAX_SIZE];
+
+    private int total = Stack.EMPTY_STACK_VALUE;
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see edu.kis.vh.nursery.Stack#getTotal()
+     */
+    @Override
+    public int getTotal() {
+        return total;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see edu.kis.vh.nursery.Stack#countIn(int)
+     */
+    @Override
+    public void countIn(final int in) {
+        if (!isFull()) {
+            numbers[total++] = in;

```

Comment:
Changing EMPTY_STACK_VALUE value requires such fixes? It does not make sense. Incorrect fix.